plugins/woocommerce/src/Internal/RestApi/Routes/V4/Settings/Tax/Controller.php

```diff
@@ -0,0 +1,417 @@
+	 *
+	 * @var WC_Settings_Tax
+	 */
+	protected $settings_tax_instance;
+
+	/**
+	 * Schema instance.
+	 *
+	 * @var TaxSettingsSchema
+	 */
+	protected $schema;
+
+	/**
+	 * Initialize the controller.
+	 *
+	 * @param TaxSettingsSchema $schema Schema class.
+	 * @internal
+	 */
+	final public function init( TaxSettingsSchema $schema ) {
+		$this->schema = $schema;
+	}
+
+	/**
+	 * Get the WC_Settings_Tax instance.
+	 *
+	 * @return WC_Settings_Tax
+	 */
+	private function get_settings_tax_instance() {
+		if ( is_null( $this->settings_tax_instance ) ) {
+			$this->settings_tax_instance = new WC_Settings_Tax();
+		}
+		return $this->settings_tax_instance;
+	}
+
+	/**
+	 * Register routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+				),
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'update_item' ),
+					'permission_callback' => array( $this, 'update_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::EDITABLE ),
+				),
+				'schema' => array( $this, 'get_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Check permissions for reading tax settings.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return bool|WP_Error
+	 */
+	public function get_item_permissions_check( $request ) {
+		if ( ! wc_rest_check_manager_permissions( 'settings', 'read' ) ) {
+			return new WP_Error(
+				'rest_forbidden',
+				__( 'Sorry, you are not allowed to access tax settings.', 'woocommerce' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+		return true;
+	}
+
+	/**
+	 * Check permissions for updating tax settings.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return bool|WP_Error
+	 */
+	public function update_item_permissions_check( $request ) {
+		if ( ! wc_rest_check_manager_permissions( 'settings', 'edit' ) ) {
+			return new WP_Error(
+				'rest_forbidden',
+				__( 'Sorry, you are not allowed to edit tax settings.', 'woocommerce' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+		return true;
+	}
+
+	/**
+	 * Get tax settings.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function get_item( $request ) {
+		try {
+			$settings = $this->get_all_settings();
+		} catch ( \Exception $e ) {
+			return new WP_Error(
+				'woocommerce_rest_tax_settings_error',
+				$e->getMessage(),
+				array( 'status' => 500 )
+			);
+		}
+
+		$response = $this->get_item_response( $settings, $request );
+		return rest_ensure_response( $response );
```

Comment:
It's better to use the `$this->get_route_error_response` and `$this->prepare_item_for_response` (which will handle the `get_item_response` too) to have consistent error reporting and response schema. 

More info here: https://wooengineering.wordpress.com/2025/09/22/developing-rest-api-v4-routes/#3-centralised-error-handling

Reply:
Yes, I was using another settings controller code, but we can actually simplify it and ensure we use `prepare_item_for_response`. Addressed in 8be36c4